pkg/imagestream/cachedimagestreamgetter.go

```diff
@@ -1,52 +1,53 @@
 package imagestream
 
 import (
-	"github.com/docker/distribution/context"
-	"github.com/docker/distribution/registry/api/errcode"
-	disterrors "github.com/docker/distribution/registry/api/v2"
+	"fmt"
 
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	imageapiv1 "github.com/openshift/api/image/v1"
 
 	"github.com/openshift/image-registry/pkg/dockerregistry/server/client"
+	rerrors "github.com/openshift/image-registry/pkg/errors"
 	quotautil "github.com/openshift/image-registry/pkg/origin-common/quota/util"
 )
 
+const (
+	ErrImageStreamGetterCode          = "ImageStreamGetter:"
+	ErrImageStreamGetterUnknownCode   = ErrImageStreamGetterCode + "Unknown"
+	ErrImageStreamGetterNotFoundCode  = ErrImageStreamGetterCode + "NotFound"
+	ErrImageStreamGetterForbiddenCode = ErrImageStreamGetterCode + "Forbidden"
+)
+
 // cachedImageStreamGetter wraps a master API client for getting image streams with a cache.
 type cachedImageStreamGetter struct {
-	ctx               context.Context
 	namespace         string
 	name              string
 	isNamespacer      client.ImageStreamsNamespacer
 	cachedImageStream *imageapiv1.ImageStream
 }
 
-func (g *cachedImageStreamGetter) get() (*imageapiv1.ImageStream, error) {
+func (g *cachedImageStreamGetter) get() (*imageapiv1.ImageStream, *rerrors.Error) {
 	if g.cachedImageStream != nil {
-		context.GetLogger(g.ctx).Debugf("(*cachedImageStreamGetter).getImageStream: returning cached copy")
 		return g.cachedImageStream, nil
 	}
 	is, err := g.isNamespacer.ImageStreams(g.namespace).Get(g.name, metav1.GetOptions{})
 	if err != nil {
-		context.GetLogger(g.ctx).Errorf("failed to get image stream: %v", err)
 		switch {
 		case kerrors.IsNotFound(err):
-			return nil, disterrors.ErrorCodeNameUnknown.WithDetail(err)
+			return nil, rerrors.NewError(ErrImageStreamGetterNotFoundCode, fmt.Sprintf("%s/%s", g.namespace, g.name), err)
 		case kerrors.IsForbidden(err), kerrors.IsUnauthorized(err), quotautil.IsErrorQuotaExceeded(err):
-			return nil, errcode.ErrorCodeDenied.WithDetail(err)
+			return nil, rerrors.NewError(ErrImageStreamGetterForbiddenCode, fmt.Sprintf("%s/%s", g.namespace, g.name), err)
 		default:
-			return nil, errcode.ErrorCodeUnknown.WithDetail(err)
+			return nil, rerrors.NewError(ErrImageStreamGetterUnknownCode, fmt.Sprintf("%s/%s", g.namespace, g.name), err)
 		}
 	}
 
-	context.GetLogger(g.ctx).Debugf("(*cachedImageStreamGetter).getImageStream: got image stream %s/%s", is.Namespace, is.Name)
 	g.cachedImageStream = is
 	return is, nil
 }
 
 func (g *cachedImageStreamGetter) cacheImageStream(is *imageapiv1.ImageStream) {
-	context.GetLogger(g.ctx).Debugf("(*cachedImageStreamGetter).cacheImageStream: got image stream %s/%s", is.Namespace, is.Name)
```

Comment:
do we not still want some of this logging to be available if we turn on debug-level logging?

Reply:
Specifically, this message does not make sense since it is written with the `g.ctx` context. There are no query parameters in it. There's nothing at all except the version of golang and the instance id.